.github/workflows/create-azure-machines.ps1

```diff
@@ -46,128 +34,185 @@ param (
 Set-StrictMode -Version "Latest"
 $PSDefaultParameterValues["*:ErrorAction"] = "Stop"
 
-$osType = $Os.Split("-")[0]
+$AzureMatrixJson = Get-Content -Path .\.github\workflows\azure-matrix.json | ConvertFrom-Json
+$FullMatrixJson = Get-Content -Path .\.github\workflows\processed-matrix.json | ConvertFrom-Json
```

Comment:
We can do it in a follow up PR, but these paths should be input parameters.

Reply:
So these values are meant to be fixed. They take the output of 'manage-azure-vms' script, which does the load-balancing and generation of the runner IDs, and it's where we will implement the Abstract Pool Queue.

Reply:
We can discuss more in follow up PRs :)